.circleci/config.yml

```
@@ -895,7 +895,7 @@ jobs:
                       <<: *CacheKeyUberjar
                   - steps: << parameters.before-steps >>
                 command: |
-                  run test-cypress-no-build <<# parameters.only-single-database >> --folder << parameters.test-files-location >> <</ parameters.only-single-database >>
+                  run test-cypress-no-build <<# parameters.test-files >> --spec << parameters.test-files >> <</ parameters.test-files >> --folder << parameters.source-folder >>
```

Just wanna make sure. There should always be `--folder`, while `--spec` should only be there if specific files are called?

---

You can use either.
Wnt for the `--folder` flag as the default one. My original idea was to use `--spec` as the default one to be able to arbitrarily combine files from different folders and group them together. Ended up reorganizing files/folders, bu there may still be the need for multiple `--spec` arguments in the future.

You can make a group called `permissions` and pass it a glob for `test-files`:
`/frontend/test/metabase/scenarios/**/permissions.cy.spec.js`

Or you can combine two files from different folders, like in the example from https://github.com/metabase/metabase/pull/15917

---

Okay, so they cannot be used together - it's either or?
Then you wanna do this:
```suggestion
                  run test-cypress-no-build <<# parameters.test-files >> --spec << parameters.test-files >> <</ parameters.test-files >> <<# parameters.source-folder >> --folder << parameters.source-folder >> <</ parameters.source-folder >>
```

---

They can even be used together.
Folder points to the source and specs can still select individual files only from that folder.